core/ordering/cosipbft/blockstore/mem.go

```diff
@@ -165,23 +158,106 @@ func (s *InMemory) WithTx(txn store.Transaction) BlockStore {
 	return store
 }
 
+// Observer is an observer that can be added to store watcher. It will announce
+// the blocks in order and without blocking the watcher even if the listener is
+// not actively emptying the queue.
+//
+// - implements core.Observer.
 type observer struct {
-	ch chan types.BlockLink
+	sync.Mutex
+
+	working sync.WaitGroup
```

Comment:
is this wait group necessary ?

Reply:
No that was for the tests, but ended up not necessary.